### DISCUSSION TITLE
Comments too dim in nvim

### DISCUSSION THREAD

**Discussion by cyaconi** [kovidgoyal/kitty#6289] [Q&A] [answered]
Is there any way to configure comments opacity in kitty? They are almost unreadable on my M1 (Retina) and it's not related to the current theme since same happens with different ones. Here's a comparison using same theme in kitty and iterm2:

**Kitty**
![image](https://github.com/kovidgoyal/kitty/assets/91154/c08ebc69-d8ce-4029-8f24-6b1c27d2f556)

**iTerm2**
<img width="455" alt="image" src="https://github.com/kovidgoyal/kitty/assets/91154/c373d72a-1fd6-4292-aed1-e6eff754521c">

in my kitty.conf I have

`background_opacity 0.98`

Thanks!

**Comment by kovidgoyal** [comment]
terminal emulators dont know anything about comments. What color they
are is set entirely by whatever application is drawing them. Change the
color scheme in whatever application you are using.

And the difference between iterm and kitty is because you are using
background blur in iterm.

**Reply by jottr** [reply]
There seems to be something off in the way kitty displays/renders colors.
Or neovim renders colors inside of kitty. 

I have the tokyonight color scheme configured in all of my editors. 
In neovim I use [tokyonight.nvim](https://github.com/folke/tokyonight.nvim#-overriding-colors--highlight-groups). 
This is what a python function looks like for me in neovim running in a kitty session:
![image](https://github.com/kovidgoyal/kitty/assets/2744198/baf0b95f-0781-4e95-ae31-fb506ca3b1ea)

Pretty much unreadable.

The same file in neovim in an iTerm session:
![image](https://github.com/kovidgoyal/kitty/assets/2744198/1f4209c8-0133-430c-84a1-21d6fb2624a3)

Which seems to be a more accurate rendering, comparing it with the original Tokyonight VSCode Theme:

![image](https://github.com/kovidgoyal/kitty/assets/2744198/9dddd8ad-58de-4a31-83b8-2b8d4a446206)

@kovidgoyal maybe you have an idea what could be causing this in kitty?

**Reply by kovidgoyal** [reply]
Again, kitty does not set colors, the application does. You need to ask the applications developers why they are outputting different colors in different terminals. Usually it will be because of some heuristic based on TERM that makes them incorrectly believe the terminal supports only 256 colors, or similar.

**Comment by cyaconi** [comment]
I've tried many different themes and it happens in most of them. Is there any config I can tinkering with to try improving readability? 

I disabled background_opacity and doesn't improve much:

<img width="424" alt="image" src="https://github.com/kovidgoyal/kitty/assets/91154/4a684c94-a8a2-4a2e-900d-fc25735f8d18">

**Reply by kovidgoyal** [reply, marked as answer]
There is no config in kitty that can change what colors an application sets.

**Comment by truongvan1003** [comment]
@kovidgoyal  Can you add a new feature to handle backgound blur? I hopy you can take time to develop this feature.
